clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -706,21 +720,24 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 shard->prev_entry_.marker.marker_type == TRACE_MARKER_TYPE_RSEQ_ABORT) {
                 saw_rseq_abort = true;
             } else {
-                // TODO i#3937: We need to exclude this check for "kernel_xfer_app" when
-                // running online. Online traces are missing instructions after sigreturns
-                // which causes the PC discontinuity checks to be violated.
-                if (knob_test_name_ != "kernel_xfer_app" || knob_offline_) {
```

Comment:
Hmm, why was this checking the test name for offline?  Was this a bug?

Reply:
Wait I think there is something weird about your sequence of commits: I'm looking at changes since my last review, but the code on the right is what's checked in.  So this is some kind of merge fix where something wasn't merged right before?